### PR TITLE
ci: include SLT tests in test/sqllogictest/transform/fold_vs_dataflow

### PR DESCRIPTION
**Pull request by nrainer-materialize**
https://materializeinc.slack.com/archives/C01LKF361MZ/p1701773762095249

SLT build: https://buildkite.com/materialize/sql-logic-tests/builds/5759

### CODE REVIEW

**Comment by nrainer-materialize** [comment]
The SLT build is failing, need to fix these before merging.

**Comment by nrainer-materialize** [comment]
@aalexandrov: Can you tell if the changes are expected? (CC @vmarcos)

File: test/sqllogictest/transform/fold_vs_dataflow/4_text_folding.slt [hunk 1]
```diff
@@ -25,18 +25,21 @@ SELECT * FROM t_data
...
+abcdef
 
 query T multiline
 EXPLAIN SELECT * FROM t_data
 WHERE f1 ILIKE '%bCd%';
 ----
 Explained Query:
-  Filter "%bCd%" ~~(#0)
-    Get materialize.public.t_data
+  Filter ilike["%bCd%"](#0)
```
Comment (ggevay) [inline test/sqllogictest/transform/fold_vs_dataflow/4_text_folding.slt:38]: The `ilike` changes are fine: their EXPLAIN format was changed in https://github.com/MaterializeInc/materialize/pull/20602

File: test/sqllogictest/transform/fold_vs_dataflow/1_numbers_dataflow.slt [hunk 2]
```diff
@@ -36,7 +36,7 @@ query T multiline
 EXPLAIN SELECT * FROM t_using_dataflow_rendering;
 ----
 Explained Query:
-  Get materialize.public.t_using_dataflow_rendering
+  ReadStorage materialize.public.t_using_dataflow_rendering
```
Comment (ggevay) [inline test/sqllogictest/transform/fold_vs_dataflow/1_numbers_dataflow.slt:39]: The `Get` -> `ReadStorage` changes are fine (from https://github.com/MaterializeInc/materialize/pull/21571).

File: test/sqllogictest/transform/fold_vs_dataflow/3_number_aggfns_dataflow.slt [hunk 3]
```diff
@@ -55,7 +55,7 @@ FROM t_using_dataflow_rendering;
 Explained Query:
   Return
     Project (#0..=#17, #30..=#35)
-      Map ((#18 / bigint_to_real(case when (#19 = 0) then null else #19 end)), (#20 / bigint_to_double(case when (#21 = 0) then null else #21 end)), (numeric_to_double(#22) / bigint_to_double(case when (#23 = 0) then null else #23 end)), (#24 / bigint_to_real(case when (#25 = 0) then null else #25 end)), (#26 / bigint_to_double(case when (#27 = 0) then null else #27 end)), (numeric_to_double(#28) / bigint_to_double(case when (#29 = 0) then null else #29 end)))
+      Map ((#18 / bigint_to_real(case when (#19 = 0) then null else #19 end)), (#20 / bigint_to_double(case when (#21 = 0) then null else #21 end)), (#22 / bigint_to_numeric(case when (#23 = 0) then null else #23 end)), (#24 / bigint_to_real(case when (#25 = 0) then null else #25 end)), (#26 / bigint_to_double(case when (#27 = 0) then null else #27 end)), (#28 / bigint_to_numeric(case when (#29 = 0) then null else #29 end)))
```
Comment (ggevay) [inline test/sqllogictest/transform/fold_vs_dataflow/3_number_aggfns_dataflow.slt:58]: I think this is this change: https://github.com/MaterializeInc/materialize/pull/15612 Can you confirm, @sploiselle?  

(Context: this is a test that has not been running in CI for a long time, and we are trying to make it run now, but some things have changed, so we are doing some archaeology to figure out if these changes are ok.)

File: test/sqllogictest/transform/fold_vs_dataflow/3_number_aggfns_dataflow.slt [hunk 4]
```diff
@@ -37,7 +37,7 @@ SELECT
  AVG(real1 + real1), AVG(double1 + double1), AVG(numeric1 + numeric1)
 FROM t_using_dataflow_rendering;
 ----
--9223372036854775808  -9223372036854775808  100000000000000000000000000000000000002  -9223372036854775808  -9223372036854775808  200000000000000000000000000000000000004  2  2  2  4  4  4  9223372036854775807  9223372036854775807  100000000000000000000000000000000000000  9223372036854775807  9223372036854775807  200000000000000000000000000000000000000  -9223372036854775808  -9223372036854775808  9223372036854775807  -9223372036854775808  -9223372036854775808  9223372036854775807
+-9223372036854775808  -9223372036854775808  100000000000000000000000000000000000002  -9223372036854775808  -9223372036854775808  200000000000000000000000000000000000004  2  2  2  4  4  4  9223372036854775807  9223372036854775807  100000000000000000000000000000000000000  9223372036854775807  9223372036854775807  200000000000000000000000000000000000000  -9223372036854775808  -9223372036854775808  50000000000000000000000000000000000001  -9223372036854775808  -9223372036854775808  100000000000000000000000000000000000002
```
Comment (ggevay) [inline test/sqllogictest/transform/fold_vs_dataflow/3_number_aggfns_dataflow.slt:40]: I guess this is somehow due to the plan change around the numeric casts, but I haven't looked further into it.

Comment (ggevay) [reply test/sqllogictest/transform/fold_vs_dataflow/3_number_aggfns_dataflow.slt:40]: The changes are matching the folding result changes, right? If yes, then it's probably ok.

**Pull request merged**